lib/persist-state.c

```diff
@@ -266,6 +304,16 @@ _alloc_value(PersistState *self, guint32 orig_size, gboolean in_use, guint8 vers
   persist_state_unmap_entry(self, self->current_ofs);
 
   self->current_ofs += size + sizeof(PersistValueHeader);
+
+  if (!_check_watermark(self) && !_grow_store(self, self->current_size + PERSIST_FILE_INITIAL_SIZE))
+    {
+      msg_error("Can't preallocate space for persist file",
+                evt_tag_int("current", self->current_size),
+                evt_tag_int("new_size", self->current_size + PERSIST_FILE_INITIAL_SIZE));
+      if (self->error_handler.handler)
```

Comment:
Maybe it would be nice to have an inline wrapper for this like: `persist_state_error_handler_run(self)`

Reply:
Thanks, indeed I pulled out to a function. Though I did not use inline, let's leave to the compiler to decide that.